networkx/algorithms/graph_hashing.py

```diff
@@ -4,10 +4,36 @@
 For now, only Weisfeiler-Lehman hashing is implemented.
 """
 
-from collections import Counter
+from collections import Counter, defaultdict
+from copy import Error
 from hashlib import blake2b
 
-__all__ = ["weisfeiler_lehman_graph_hash"]
+__all__ = ["weisfeiler_lehman_graph_hash", "weisfeiler_lehman_subgraph_hashes"]
+
+
+def _hash_label(label, digest_size):
+    return blake2b(label.encode("ascii"), digest_size=digest_size).hexdigest()
+
+
+def _init_node_labels(G, edge_attr, node_attr):
+    if node_attr:
+        return {u: str(G.nodes[u][node_attr]) for u in G.nodes()}
+    elif edge_attr:
+        return {u: "" for u in G.nodes()}
+    else:
+        return {u: str(G.degree(u)) for u in G.nodes()}
```

Comment:
```suggestion
        return {u: str(val) for u, val in G.nodes(data=node_attr)}
    elif edge_attr:
        return {u: "" for u in G}
    else:
        return {u: str(deg) for u, deg in G.degree()}
```

Reply:
an unwanted side effect of 
```
{u: str(val) for u, val in G.nodes(data=node_attr)}
```
is that when a node has no node attribute defined, this will give the string `'None'` in this dictionary, whereas previously this raised a `KeyError`, which I think is helpful behaviour for the user (gives them the responsibility for filling default values for node attributes. Otherwise graphs could pass isomorphism tests unexpectedly if enough nodes have missing attributes

Reply:
Yeah, it's only on initialising labels too so not a bottleneck - but there's no reason not to use your elegant suggestion, so I've changed it